neurodamus/stimulus_manager.py

```diff
@@ -342,12 +342,30 @@ def params_from_mean_var(self, mean, var):
         # bi-exponential time to peak [ms]
         t_peak = log(self.tau_D / self.tau_R) / (1 / self.tau_R - 1 / self.tau_D)
         # bi-exponential peak height [1]
-        x_peak = exp(-t_peak / self.tau_D) - exp(-t_peak / self.tau_R)
+        F_peak = exp(-t_peak / self.tau_D) - exp(-t_peak / self.tau_R)
 
-        rate_ms = (1 + self.cv_square) / 2 * (mean ** 2 / var) / (self.tau_D + self.tau_R)
-        self.rate = rate_ms * 1000  # rate in 1 / s [Hz]
-        self.amp_mean = mean * x_peak / rate_ms / (self.tau_D - self.tau_R)
-        self.amp_var = self.cv_square * self.amp_mean ** 2
+        # utility constants
+        Xi = (self.tau_D - self.tau_R) / F_peak
+        A = 1 / (self.tau_D + self.tau_R)
+        B = 1 / ((self.tau_D + 2 * self.tau_R) * (2 * self.tau_D + self.tau_R))
+
+        # skewness
+        skew_bnd_min = (8 / 3) * (B / A ** 2) * (sd / mean)
+        skew = (1 + self.rel_skew) * skew_bnd_min
+        if skew < skew_bnd_min or skew > 2 * skew_bnd_min:
+            raise Exception("%s skewness out of bounds" % self.__class__.__name__)
+
+        # cumulants
+        lambda2_1 = sd ** 2 / mean                     # lambda2 over lambda1
+        lambda3_2 = sd * skew                          # lambda3 over lambda2
+        theta1pk = 2 / (A * Xi) * lambda2_1            # = (1 + k) * theta
+        theta2pk = (3 * A) / (4 * B * Xi) * lambda3_2  # = (2 + k) * theta
+
+        # derived parameters
+        self.amp_mean = 2 * theta1pk - theta2pk               # mean amplitude [nA or uS]
+        self.amp_var = self.amp_mean * (theta2pk - theta1pk)  # variance of amplitude [nA^2 or uS^2]
```

Comment:
Why A _or_ S? Don't you need a voltage to convert between these two units?

Reply:
Hi @matz-e . It's one or the other, so no conversion is necessary. When the signal is injected as a Current, it has units of A, and when injected as a Conductance it has units of S. :)

Reply:
Of course, when it's injected as a conductance, there is a voltage value involved (the holding voltage of the voltage clamp used to implement the dynamic clamp) to convert to a membrane current.

Reply:
Thank you for clarifying!